### PR TITLE
ZipkinExporter for net452 will export tags as string

### PR DESCRIPTION
**Pull request by eddynaka**
https://github.com/open-telemetry/opentelemetry-dotnet/pull/1168#discussion_r477451513

cc: @alanwest 

## Changes
- some PR's ago, I changed the ZipkinExporter for other frameworks to export as string. Now, I'm adding the same functionality to net452.

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs [hunk 1]
```diff
@@ -191,12 +192,35 @@ public void Write(JsonWriter writer)
...
+                        case bool[] boolArrayValue:
+                            writer.WriteValue(string.Join(",", boolArrayValue));
+                            break;
+                        default:
+                            writer.WriteValue(tag.Value.ToString());
+                            break;
+                    }
                 }
 
+                Thread.CurrentThread.CurrentUICulture = originalUICulture;
```
Comment (alanwest) [inline src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs:222]: Should we be defensive here? If the above code fails for some reason we may end up not setting the `CurrentUICulture` back to the previous value.

File: test/OpenTelemetry.Exporter.Zipkin.Tests/ZipkinExporterTests.cs [hunk 2]
```diff
@@ -229,7 +229,7 @@ public void ZipkinExporterIntegrationTest(bool useShortTraceIds)
             var activitySource = new ActivitySource(nameof(CreateTestActivity));
 
             var tags = setAttributes ?
-                    attributes.Select(kvp => new KeyValuePair<string, object>(kvp.Key, kvp.Value.ToString()))
```
Comment (alanwest) [inline test/OpenTelemetry.Exporter.Zipkin.Tests/ZipkinExporterTests.cs:232]: 🎉

File: src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs (hunk 1 above)
Comment (eddynaka) [reply src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs:222]: sure. i can add a Try/Finally in both cases

Comment (eddynaka) [reply src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs:222]: done.

**Comment by alanwest** [comment]
LGTM

File: src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs [hunk 3]
```diff
@@ -191,10 +192,38 @@ public void Write(JsonWriter writer)
                 writer.WritePropertyName("tags");
                 writer.WriteStartObject();
 
-                foreach (var tag in this.Tags.Value)
+                // this will be used when we convert int, double, int[], double[] to string
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs:195]: This looks familiar, when we did it before was it in a directive for netcoreapp or netstandard or something?

Comment (eddynaka) [reply src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs:195]: Yes. The same file, but for net46, standard that can use system.text.json

Comment (eddynaka) [reply src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs:195]: To facilitate, i pushed a new version where i create a helper function. This function will be used in all places to convert an object to string based on its type.

File: src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs [hunk 4]
```diff
@@ -317,5 +317,18 @@ public void Write(Utf8JsonWriter writer)
         }
 
 #endif
+
+        private string ConvertObjectToString(object obj)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs:321]: Nice refactor! Let's add `AggressiveInlining` hint to this?

Comment (eddynaka) [reply src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs:321]: done!

**Pull request merged**